documentation/projects/proposals/rekognition_data/20240423-implementation_plan_machine_generated_tags_api.md

```diff
@@ -0,0 +1,331 @@
+searches are made in Elasticsearch. It will not cover the semantics of the
+actual labels (i.e. which machine-generated labels to retain or discard), as
+that is the purview of #4040.
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+When this work is completed, it is expected that we will be able to clearly
+distinguish which tags returned by the API are machine-generated and where those
+tags came from. We will also have a clearly documented approach for handling
+duplicate tags and how this affects search results.
+
+## Step-by-step plan
+
+<!--
+List the ordered steps of the plan in the form of imperative-tone issue titles.
+
+The goal of this section is to give a high-level view of the order of implementation any relationships like
+blockages or other dependencies that exist between steps of the plan. Link each step to the step description
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+The implementation of this work involves two primary pieces:
+
+1. [Expose provider information for each tag within the results](#expose-provider-in-tags)
+2. [Apply modifications (if necessary) to the Elasticsearch document scoring](#modify-elasticsearch-document-scoring)
+
+## Step details
+
+<!--
+Describe all of the implementation steps listed in the "step-by-step plan" in detail.
+
+For each step description, ensure the heading includes an obvious reference to the step as described in the
+"step-by-step plan" section above.
+-->
+
+### Expose provider in tags
+
+This step will be straightforward, as the provider information for tags already
+exists in the API database. In order to have them show up in the API results, we
+would need to modify the
+[`TagSerializer`](https://github.com/WordPress/openverse/blob/3ed38fc4b138af2f6ac03fcc065ec633d6905d73/api/api/serializers/media_serializers.py#L442)
+to include the `provider` field as well. This would change the returned `tags`
+entry for a result from:
+
+```json
+[
+  {
+    "name": "businesscard",
+    "accuracy": null
+  },
+  {
+    "name": "design",
+    "accuracy": null
+  },
+  {
+    "name": "projects",
+    "accuracy": null
+  },
+  {
+    "name": "achievement",
+    "accuracy": 0.94971
+  },
+  {
+    "name": "box",
+    "accuracy": 0.90665
+  },
+  {
+    "name": "business",
+    "accuracy": 0.9916
+  },
+  {
+    "name": "card",
+    "accuracy": 0.94793
+  }
+]
+```
+
+to the following:
+
+```json
+[
+  {
+    "name": "businesscard",
+    "accuracy": null,
+    "provider": "flickr"
+  },
+  {
+    "name": "design",
+    "accuracy": null,
+    "provider": "flickr"
+  },
+  {
+    "name": "projects",
+    "accuracy": null,
+    "provider": "flickr"
+  },
+  {
+    "name": "achievement",
+    "accuracy": 0.94971,
+    "provider": "flickr"
```

Comment:
Nit: technically `accuracy` should be `null` or the `provider` should be `clarifai`

Reply:
Whoops, good catch!